R/database.R

```diff
@@ -1715,98 +1715,154 @@ get_sample_cn_segments = function(this_sample_id,
-#'
-get_cn_segments = function(chromosome = "",
+#' #Example using chromosome, qstart and qend parameters:
+#' segments_region_grch37 = get_cn_segments(chromosome = "chr8",
+#'                                          qstart = 128723128,
+#'                                          qend = 128774067,
+#'                                          projection = "grch37", 
+#'                                          this_seq_type = "genome", 
+#'                                          remove_chr_prefix = FALSE, 
+#'                                          from_flatfile = TRUE, 
+#'                                          streamlined = FALSE) 
+#'                                     
+#' #Example using the regions parameter:
+#' segments_region_hg38 = get_cn_segments(region = "chr8:128,723,128-128,774,067",
+#'                                        projection = "hg38", 
+#'                                        this_seq_type = "genome", 
+#'                                        remove_chr_prefix = TRUE, 
+#'                                        from_flatfile = TRUE, 
+#'                                        streamlined = FALSE) 
+#'
+#'
+get_cn_segments = function(region,
+                           chromosome,
                            qstart,
                            qend,
-                           region,
-                           with_chr_prefix = FALSE,
+                           projection = "grch37",
+                           this_seq_type = "genome",
+                           remove_chr_prefix = FALSE,
                            streamlined = FALSE,
-                           from_flatfile = FALSE){
+                           from_flatfile = TRUE){
+  
+  #checks
   remote_session = check_remote_configuration()
-  db = config::get("database_name")
-  table_name = config::get("results_tables")$copy_number
-  table_name_unmatched = config::get("results_tables")$copy_number_unmatched
+  
+  #check seq type and return a message if anything besides "genome" is called. To be updated once capture samples have been processed through CNV protocols.
+  if(this_seq_type!="genome"){
+    stop("Currently, only genome samples are available for this function. Please select a valid seq type (i.e genome). Compatibility for capture samples will be added soon...")
+  }
+  
+  #get wildcards from this_seq_type (lazy)
+  seq_type = this_seq_type
+  
+  #perform wrangling on the region to have it in the correct format. 
   if(!missing(region)){
     region = gsub(",", "", region)
-    #format is chr6:37060224-37151701
     split_chunks = unlist(strsplit(region, ":"))
     chromosome = split_chunks[1]
     startend = unlist(strsplit(split_chunks[2], "-"))
     qstart = startend[1]
     qend = startend[2]
   }
-  if(grepl("chr", chromosome)){
+  
+  #deal with chr prefixes for region, based on selected genome projection.
+  if(projection == "grch37"){
+    if(grepl("chr", chromosome)){
```

Comment:
I think this ifelse is not needed. The `gsub` will not complain when the "chr" is not in the string. Instead, the user here will get warning that only first element is used since the `chromosome` is a vector. I think this part can be reworked to something like
```
always gsub chr substring
if(projection == "hg38"){
    add the chr prefix back using paste0
}
```
This way, there are no ifelse checks so the code is simpler, prefix is properly handled, and user will get no warnings

Reply:
The else part is needed because it's prepending chr if (and only if ) it's missing though

Reply:
Right, I meant that it is only checking the first occurrence. Here it is str_detect but grepl I think works the same way. So If there will be a mix of prefixed and non-prefixed chromosome names, the way it is currently implemented might not always deal with this properly
```
Warning message:
In if (!str_detect(genes$chromosome, "chr")) { :
  the condition has length > 1 and only the first element will be used
```